src/OpenTelemetry.Api/Context/Propagation/PropagationContext.cs

```diff
@@ -24,24 +24,32 @@ namespace OpenTelemetry.Context.Propagation
     /// </summary>
     public readonly struct PropagationContext : IEquatable<PropagationContext>
     {
+        private readonly ActivityContext activityContext;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="PropagationContext"/> struct.
         /// </summary>
         /// <param name="activityContext"><see cref="System.Diagnostics.ActivityContext"/>.</param>
-        /// <param name="baggage"><see cref="Baggage"/>.</param>
+        /// <param name="baggage"><see cref="OpenTelemetry.Baggage"/>.</param>
         public PropagationContext(ActivityContext activityContext, Baggage baggage)
         {
-            this.ActivityContext = activityContext;
+            this.activityContext = activityContext;
+            this.Baggage = baggage;
+        }
+
+        private PropagationContext(Activity activity, Baggage baggage)
+        {
+            this.activityContext = activity?.Context ?? default;
             this.Baggage = baggage;
         }
 
         /// <summary>
         /// Gets <see cref="System.Diagnostics.ActivityContext"/>.
         /// </summary>
-        public ActivityContext ActivityContext { get; }
+        public ActivityContext ActivityContext => this.activityContext;
```

Comment:
Should we mark this obsolete as well to encourage the use of `GetActivityContextRef`?